### PR TITLE
Clarify serializer usage

### LINKED ISSUE #254: Clarify serializers usage
The [serializer implementations doc](https://docs.axoniq.io/reference-guide/axon-framework/serialization#serializer-implementations) should be enriched to include more usages and be more precise about what each serializer is supposed to handle.

This is related to [this issue](https://github.com/AxonFramework/AxonFramework/issues/2229), where @smcvb confirmed to me that query `ResponseType`s were indeed meant to be handled by the default serialized rather than the one meant for message payloads.

### PR DESCRIPTION
**Pull request by ghost**
closes #254

### CODE REVIEW

**Review by gklijs — APPROVED** [review]
Looks good to me 👍

File: axon-framework/serialization.md [hunk 1]
```diff
@@ -20,21 +20,22 @@ You may also implement your own serializer simply by creating a class that imple
...
 
-2. The Message `Serializer` is in charge of de-/serializing the command and query messages \(used in a distributed application setup\).
+2. The Message `Serializer` is in charge of (de)serializing the command and query message payload and metadata \(used in a distributed application setup\).
    Messages are shared between nodes and typically need to be interoperable and/or compact.
    Take this into account when choosing the message serializer implementation.
 
-3. The default `Serializer` is in charge of de-/serializing the remainder, being the tokens, snapshots and sagas.
+3. The default `Serializer` is in charge of (de)serializing the remainder, being the messages (except the payload and metadata), tokens, snapshots, deadlines and sagas.
    These objects are generally not shared between different applications, and most of these classes aren't expected to have some of the getters and setters that are, for example, typically required by Jackson based serializers.
+   For example a QueryMessage consists of a ResponseType, which will be (de)serialized using the `default`serializer, the query request and response payload will be (de)serialized using the`message`serializer.
```
Comment (smcvb) [inline axon-framework/serialization.md:38]: Minor suggestion here:

```suggestion
   For example, a `QueryMessage` consists of a payload and `ResponseType`, which will respectively be (de)serialized using the `message` and the `default` serializer.
```

**Review by smcvb — APPROVED** [review]
Small suggestion on the response type bit. Otherwise this looks good to me.

**Review by CodeDrivenMitch — APPROVED** [review]
Looks good to me

**Pull request merged**